src/session/session_api.c

```diff
@@ -488,6 +488,57 @@ err:	API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
 /*
+ * __session_rename --
+ *	WT_SESSION->reset method.
+ */
+static int
+__session_reset(WT_SESSION *wt_session, const char *config)
+{
+	WT_CONNECTION_IMPL *conn;
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_SESSION_IMPL *session;
+
+	conn = (WT_CONNECTION_IMPL *)wt_session->connection;
+	session = (WT_SESSION_IMPL *)wt_session;
+
+	SESSION_API_CALL(session, reset, config, cfg);
+
+	if (F_ISSET(&session->txn, WT_TXN_RUNNING))
+		WT_ERR_MSG(session, EINVAL, "transaction in progress");
```

Comment:
Hi,

I vote with Michael on this one, and it's consistent with what we do in
other APIs, including WT_SESSION::reconfigure and
WT_SESSION::transaction_sync

- Alex

On Tue, Jun 23, 2015 at 8:58 AM, Michael Cahill <notifications@github.com>
wrote:

> In src/session/session_api.c
> <https://github.com/wiredtiger/wiredtiger/pull/2022#discussion_r32993771>:
>
> > + */
> > +static int
> > +__session_reset(WT_SESSION *wt_session, const char *config)
> > +{
> > +	WT_CONNECTION_IMPL *conn;
> > +	WT_CURSOR *cursor;
> > +	WT_DECL_RET;
> > +	WT_SESSION_IMPL *session;
> > +
> > +	conn = (WT_CONNECTION_IMPL *)wt_session->connection;
> > +	session = (WT_SESSION_IMPL *)wt_session;
> > +
> > +	SESSION_API_CALL(session, reset, config, cfg);
> > +
> > +	if (F_ISSET(&session->txn, WT_TXN_RUNNING))
> > +		WT_ERR_MSG(session, EINVAL, "transaction in progress");
>
> I think it's worse to allow this and have folks write bad application
> code. This error will ensure that using WT_SESSION::reset is caught during
> development. I'm much more comfortable with that than with having
> applications call WT_SESSION::reset when they actually want to roll back a
> running transaction.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wiredtiger/wiredtiger/pull/2022/files#r32993771>.
>